src/util/AbstractMemory/Memory.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright 2022, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (July of 2023,
+// schlegea@informatik.uni-freiburg.de)
+
+#include <absl/strings/str_cat.h>
+
+#include <string_view>
+#include <tuple>
+
+#include "util/AbstractMemory/Memory.h"
+#include "util/AbstractMemory/MemoryDefinitionLanguageVisitor.h"
+#include "util/AbstractMemory/generated/MemoryDefinitionLanguageLexer.h"
+#include "util/AbstractMemory/generated/MemoryDefinitionLanguageParser.h"
+#include "util/antlr/ANTLRErrorHandling.h"
+
+namespace ad_utility {
+// _____________________________________________________________________________
+Memory& Memory::operator=(const size_t& amountOfMemoryInBytes) {
+  memoryInBytes_ = amountOfMemoryInBytes;
+  return *this;
+}
+
+// _____________________________________________________________________________
+Memory& Memory::operator=(size_t&& amountOfMemoryInBytes) {
+  memoryInBytes_ = std::move(amountOfMemoryInBytes);
+  return *this;
+}
+
+// _____________________________________________________________________________
+size_t Memory::bytes() const { return memoryInBytes_; }
+
+// Helper function for dividing two instances of `size_t`. Mainly exists for
+// code reduction.
+static double sizetDivision(const size_t& dividend, const size_t& divisor) {
+  auto dv = std::lldiv(dividend, divisor);
+  return static_cast<double>(dv.quot) + static_cast<double>(dv.rem) / divisor;
+}
+
+// _____________________________________________________________________________
+double Memory::kilobytes() const {
+  return sizetDivision(memoryInBytes_, numberOfBytesPerKB);
```

Comment:
What I've been meaning to tell you for some time:
you can always use `num` instead of `numberOf`, so `numBytesPerKB` is completely sufficient (same for all other of these variables and all `numRows` and `numCols`.

Reply:
Done, see [fd5dcd8](https://github.com/ad-freiburg/qlever/pull/1032/commits/fd5dcd8a673800347af946f0c623e77ef4f0aa8b).